CameraX-MLKit/app/src/main/java/com/example/camerax_mlkit/QrCodeDrawable.kt

```diff
@@ -0,0 +1,93 @@
+package com.example.camerax_mlkit
+
+import android.content.Intent
+import android.graphics.*
+import android.graphics.drawable.Drawable
+import android.net.Uri
+import android.view.MotionEvent
+import android.view.View
+import com.google.mlkit.vision.barcode.common.Barcode
+
+
+class QrCodeDrawable(barcode: Barcode) : Drawable() {
+    private val boundingRectPaint = Paint().apply {
+        style = Paint.Style.STROKE
+        color = Color.YELLOW
+        strokeWidth = 5F
+        alpha = 200
+    }
+
+    private val contentRectPaint = Paint().apply {
+        style = Paint.Style.FILL
+        color = Color.YELLOW
+        alpha = 255
+    }
+
+    private val contentTextPaint = Paint().apply {
+        color = Color.DKGRAY
+        alpha = 255
+        textSize = 36F
+    }
+
+    private var boundingRect: Rect = barcode.boundingBox!!
+    private val contentPadding = 25
+
+    private var qrContent: String = ""
+    private var textWidth: Int = 0
+    var qrCodeTouchCallback = { v: View, e: MotionEvent -> false} //no-op
+
+    init {
+        when (barcode.valueType) {
+            Barcode.TYPE_URL -> {
+                qrContent = barcode.url!!.url!!
+                qrCodeTouchCallback = { v: View, e: MotionEvent ->
+                    if (e.action == MotionEvent.ACTION_DOWN && boundingRect.contains(e.getX().toInt(), e.getY().toInt())) {
+                        val openBrowserIntent = Intent(Intent.ACTION_VIEW)
+                        openBrowserIntent.data = Uri.parse(qrContent)
+                        v.context.startActivity(openBrowserIntent)
+                    }
+                    true // return true from the callback to signify the event was handled
+                }
+            }
+            // Add other QR Code types here to handle other types of data,
+            // like Wifi credentials.
+            else -> {
+                qrContent = "Unrecognized data type"
```

Comment:
Can we show the value of [barcode.getRawValue](https://developers.google.com/android/reference/com/google/mlkit/vision/barcode/common/Barcode#public-string-getrawvalue) here?  It would be more useful .